src/components/header/renderer.tsx

```diff
@@ -173,9 +194,24 @@ class Header extends React.Component<Props, State> {
         });
         throw Error();
       }
+      this.setState({
+        invalidFilename: false,
+      });
+    } else {
+      const xfilename = Object.keys(gistData.files).find(f => gistData.files[f].language === 'JSON');
```

Comment:
what is xfilename?

Reply:
xfilename is the string which contains the filename from gistData which has language json.
I am checking if the filename user enters and the returned filename are same or not.

Reply:
Please find a better variable name.